Zoom-on-Scroll Support in Image Preiew

Vaibhaviitian opened the pull request:
Adding feature regarding this issue #528  - Add Zoom-on-Scroll Support in Media View

This PR adds zoom in and out on scroll down and up
Wherever mouse is, the zoom or scale factor will be large for that area (as shown in the video)


https://github.com/user-attachments/assets/6c3b437e-59b2-427b-ad5e-66db844a378c

rahulharpal1603 commented:
@Vaibhaviitian 

Please fix the linting errors

rahulharpal1603 commented on frontend/src/components/Media/ImageCard.tsx:
```diff
@@ -32,6 +32,7 @@ export function ImageCard({
   const tags = image.tags || [];
 
   const handleImageClick = useCallback(() => {
+    console.log(imageIndex);
```
remove console log

rahulharpal1603 commented on frontend/src/pages/Home/Home.tsx:
```diff
@@ -29,6 +29,7 @@ export const Home = () => {
 
   // Handle fetching lifecycle
   useEffect(() => {
+    console.log(displayImages);
```
remove console log

rahulharpal1603 commented:
> 
> > The current approach doesn't seem very intuitive. When I have my mouse pointer on the door handle, it should zoom near that thing, not anywhere else (See the attached video)
> > 
> >  2025-10-07.00-44-31.mov
> 
> sir i had tried to make as ideal zoom work but what happens when user scrolled for first time then the x,y coordinate for the user cursor get changed and then scaling happening with respect to that new x,y so this was the issue i had faced ,
> so sir let me know if i haave to work more on it
> 

Yes, you have to improve it.

Vaibhaviitian commented:
@rahulharpal1603 sir i opened some vertical images but seems no issue see the video i had attached

Vaibhaviitian commented:
https://github.com/user-attachments/assets/6f0c8ba4-583b-413c-bc3e-040017180d48

see it sir @rahulharpal1603 , and i corrected relative

rahulharpal1603 commented:
> > > Recording.2025-10-19.131926.mp4
> > > see it sir @rahulharpal1603 , and i corrected relative
> > 
> > 
> > Don't take a screenshot of the boat image. Download its full resolution version and then show how it's being displayed. link: https://images.unsplash.com/photo-1544376798-89aa6b82c6cd?ixlib=rb-4.1.0&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&q=80&w=774
> 
> sir i had added this photo in the folder but not visible in the pictopy

Delete the PictoPy.db file and again add the folder.

Vaibhaviitian commented:
> > > > Recording.2025-10-19.131926.mp4
> > > > see it sir @rahulharpal1603 , and i corrected relative
> > > 
> > > 
> > > Don't take a screenshot of the boat image. Download its full resolution version and then show how it's being displayed. link: https://images.unsplash.com/photo-1544376798-89aa6b82c6cd?ixlib=rb-4.1.0&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&q=80&w=774
> > 
> > 
> > sir i had added this photo in the folder but not visible in the pictopy
> 
> Delete the PictoPy.db file and again add the folder.

sir i had tried this

Vaibhaviitian commented:
sir i guess the issue in the image dimension see it sir that boat image dim of 3k*4.5k and rest have less 
![WhatsApp Image 2025-10-19 at 15 14 02_bf5a0727](https://github.com/user-attachments/assets/48fb89f1-632e-4dde-8a2d-aec8aa1ed487)

above image is also vertical but haaving less dimension 


![WhatsApp Image 2025-10-19 at 15 14 03_13e3f43a](https://github.com/user-attachments/assets/3d671430-ad56-4999-a04f-eb531c933e12)
and sir this boad image have large dimension

Vaibhaviitian commented:
so sir i think it is not a problem that boat image have large dimension that is the issue if you makes it dimension shorter then it would be correct
